algosdk/atomic_transaction_composer.py

```diff
@@ -0,0 +1,529 @@
+        committed to a block. An error will be thrown if submission fails.
+        The composer's status must be SUBMITTED or lower before calling this method.
+        If submission is successful, this composer's status will update to SUBMITTED.
+
+        Note: a group can only be submitted again if it fails.
+
+        Args:
+            client (AlgodClient): Algod V2 client
+
+        Returns:
+            list[Transaction]: list of submitted transactions
+        """
+        if self.status <= AtomicTransactionComposerStatus.SUBMITTED:
+            self.gather_signatures()
+        else:
+            raise error.AtomicTransactionComposerError(
+                "AtomicTransactionComposerStatus must be submitted or lower to submit a group"
+            )
+
+        client.send_transactions(self.signed_txns)
+        self.status = AtomicTransactionComposerStatus.SUBMITTED
+        return self.tx_ids
+
+    def execute(self, client, wait_rounds):
+        """
+        Send the transaction group to the network and wait until it's committed
+        to a block. An error will be thrown if submission or execution fails.
+        The composer's status must be SUBMITTED or lower before calling this method,
+        since execution is only allowed once. If submission is successful,
+        this composer's status will update to SUBMITTED.
+        If the execution is also successful, this composer's status will update to COMMITTED.
+
+        Note: a group can only be submitted again if it fails.
+
+        Args:
+            client (AlgodClient): Algod V2 client
+            wait_rounds (int): maximum number of rounds to wait for transaction confirmation
+
+        Returns:
+            AtomicTransactionResponse: Object with confirmed round for this transaction,
+                a list of txIDs of the submitted transactions, and an array of
+                results for each method call transaction in this group. If a
+                method has no return value (void), then the method results array
+                will contain None for that method's return value.
+        """
+        if self.status > AtomicTransactionComposerStatus.SUBMITTED:
+            raise error.AtomicTransactionComposerError(
+                "AtomicTransactionComposerStatus must be submitted or lower to execute a group"
+            )
+
+        self.submit(client)
+        resp = transaction.wait_for_confirmation(
+            client, self.tx_ids[0], wait_rounds
+        )
+        self.status = AtomicTransactionComposerStatus.COMMITTED
+
+        confirmed_round = resp["confirmed-round"]
+        method_results = []
+
+        for i, tx_id in enumerate(self.tx_ids):
+            raw_value = None
+            return_value = None
+            decode_error = None
+
+            if i not in self.method_dict:
+                continue
+            # Return is void
+            if self.method_dict[i].returns.type == "void":
```

Comment:
```suggestion
            if self.method_dict[i].returns.type == RETURNS.VOID:
```

Assuming you took my `RETURNS.VOID` suggestion above....